src/QsCompiler/CompilationManager/AssemblyLoader.cs

```diff
@@ -172,6 +160,79 @@ public static bool LoadReferencedAssembly(Uri asm, out References.Headers header
+        /// <param name="assemblyPath">The path to the file.</param>
+        /// <returns>
+        /// The PEReader for the given assembly.
+        /// </returns>
+        /// <exception cref="FileNotFoundException"><paramref name="assemblyPath"/> does not exist.</exception>
+        private static PEReader GetAssemblyReader(string assemblyPath)
+        {
+            if (!File.Exists(assemblyPath))
+            {
+                throw new FileNotFoundException($"The file '{assemblyPath}' given to the assembly loader does not exist.");
+            }
+            return new PEReader(File.OpenRead(assemblyPath));
+        }
+
+        /// <summary>
+        /// Loads a resource named <paramref name="resourceName"/> from <paramref name="assemblyReader"/> into the stream <paramref name="resourceStream"/>.
+        /// </summary>
+        /// <param name="resourceName">The name of the resource to load.</param>
+        /// <param name="assemblyReader">The Portable Executable Reader for the dll from which to load the resource.</param>
+        /// <param name="resourceStream">A stream for reading the resource as a bytes.</param>
+        /// <returns>
+        /// True if <paramref name="assemblyReader"/> includes the resource named by <paramref name="resourceName"/>, false otherwise.
+        /// </returns>
+        private static bool LoadResource(string resourceName, PEReader assemblyReader, Stream resourceStream)
+        {
+            var metadataReader = assemblyReader.GetMetadataReader();
+
+            bool isResourcePresent = false;
+            ManifestResource resource;
+            if (metadataReader.Resources().TryGetValue(resourceName, out resource))
+            {
+                isResourcePresent = true;
+            }
+
+            // The offset of resources is relative to the resources directory.
+            // It is possible that there is no offset given because a valid dll allows for extenal resources.
+            // In all Q# dlls there will be a resource with the specific name chosen by the compiler.
+            var resourceDir = assemblyReader.PEHeaders.CorHeader.ResourcesDirectory;
+            if (!assemblyReader.PEHeaders.TryGetDirectoryOffset(resourceDir, out var directoryOffset) ||
+                !isResourcePresent ||
+                !resource.Implementation.IsNil)
+            {
+                return false;
+            }
+
+            var image = assemblyReader.GetEntireImage(); // uses int to denote the length and access parameters
+            var absResourceOffset = (int)resource.Offset + directoryOffset;
+
+            // the first four bytes of the resource denote how long the resource is, and are followed by the actual resource data
+            var resourceLength = BitConverter.ToInt32(image.GetContent(absResourceOffset, sizeof(int)).ToArray(), 0);
+            new MemoryStream(image.GetContent(absResourceOffset + sizeof(int), resourceLength).ToArray()).CopyTo(resourceStream);
+
```

Comment:
```suggestion
            using var tempStream = new MemoryStream(image.GetContent(absResourceOffset + sizeof(int), resourceLength).ToArray());
            tempStream.CopyTo(resourceStream);
            resourceStream.Flush();
            resourceStrea.Position = 0;
```

Reply:
Note that for the temporary memory stream the `using` keyword is introduced such that it is disposed after it has served its purpose.

Reply:
done